File: src/main/java/io/github/gravetii/scene/help/AboutScene.java
package io.github.gravetii.scene.help;

import io.github.gravetii.scene.FxDimensions;
import io.github.gravetii.scene.FxScene;
import javafx.geometry.Dimension2D;
import javafx.stage.Stage;

import java.util.Optional;

public class AboutScene extends FxScene {
  private final AboutComponent component;

  public AboutScene(Stage stage) throws Exception {
    super(stage);
    this.component = new AboutComponent();
  }

  @Override
  protected void build() {
    this.showCenter(this.component);
  }

  @Override
  protected String title() {
    return "About";
  }

  @Override
  protected Optional<FxDimensions> preferredDimensions() {
    FxDimensions dimensions = new FxDimensions(new Dimension2D(500, 200));
    return Optional.of(dimensions);
  }
}

File: src/main/kotlin/io/github/gravetii/scene/help/AboutScene.kt
package io.github.gravetii.scene.help

import io.github.gravetii.scene.FxDimensions
import io.github.gravetii.scene.FxScene
import javafx.geometry.Dimension2D
import javafx.stage.Stage

class AboutScene(stage: Stage) : FxScene(stage) {

    private val component = AboutComponent()

    override fun build() = showCenter(component)

    override fun title() = "About"

    override fun preferredDimensions(): FxDimensions = FxDimensions(Dimension2D(500.0, 200.0))

}